### ISSUE TITLE
Set serverside select value from other component is not working

### ISSUE THREAD

**Issue by CodeAn** [livewire/flux#485]
Hey,

I'm playing around with the select in serverside. Running into an issue setting the value from outside the component.

Using Flux 1.0.14, Livewire 3.5.12

Component blade file (update.blade.php):
```
<div>
<flux:button wire:click="launch">Launch</flux:button>
<flux:modal name="update" class="md:w-96 space-y-6" variant="flyout">
    <div>
        <flux:heading size="lg">Update select</flux:heading>
        <flux:subheading>Lets see what it does.</flux:subheading>
    </div>

    {{ $value}}
   
    <flux:button.group>
        <flux:select variant="listbox" wire:model="value" searchable placeholder="{{ __('Select a value...') }}" :filter="false">
            <x-slot name="search">
                <flux:select.search placeholder="Search..." wire:model="query" wire:keyup="search" />
            </x-slot>

            @foreach($options as $option)
                <flux:option value="{{ $option['id'] }}">
                    {{ $option['name'] }}
                </flux:option>
            @endforeach
        </flux:select>
        <flux:button wire:click="clear" tooltip="Clear selection" icon="x-mark"></flux:button>

    </flux:button.group>

    <div class="flex">
        <flux:spacer />

        <flux:button type="submit" variant="primary">Save changes</flux:button>
    </div>
</flux:modal>
</div>

```

Component update.php
```
<?php

namespace App\Livewire;

use Livewire\Component;
use Livewire\Attributes\On;
use Flux\Flux;

class Update extends Component
{
    public $options;
    public $value;
    public $query = '';
    public function mount()
    {
        $this->options = collect([
            [
                'id' => 1,
                'name' => 'John Doe',
            ],
            [
                'id' => 2,
                'name' => 'Jane Doe',
            ],
            [
                'id' => 3,
                'name' => 'Alice Smith',
            ],
            [
                'id' => 4,
                'name' => 'Bob Johnson',
            ],
        ]);
    }
    public function clear() {
        $this->value = null;
        $this->query = '';
        $this->options = collect([
            [
                'id' => 1,
                'name' => 'John Doe',
            ],
            [
                'id' => 2,
                'name' => 'Jane Doe',
            ],
            [
                'id' => 3,
                'name' => 'Alice Smith',
            ],
            [
                'id' => 4,
                'name' => 'Bob Johnson',
            ],
        ]);
    }
    public function search()
    {
        $searchTerms = explode(' ', $this->query);
        $this->options = collect([
            [
                'id' => 1,
                'name' => 'John Doe',
            ],
            [
                'id' => 2,
                'name' => 'Jane Doe',
            ],
            [
                'id' => 3,
                'name' => 'Alice Smith',
            ],
            [
                'id' => 4,
                'name' => 'Bob Johnson',
            ],
        ])->filter(function ($option) use ($searchTerms) {
            foreach ($searchTerms as $term) {
                if (stripos($option['name'], $term) !== false) {
                    return true;
                }
            }
            return false;
        });
        
    }
    public function launch() {
        $this->dispatch('setValue', optionId: 5);
        Flux::modal('update')->show();
    }
    #[On('setValue')]
    public function setValue($optionId)
    {
        $this->options = collect([
            
            [
                'id' => 5,
                'name' => 'John Smith',
            ]
            ]);
        $this->value = $optionId;
        $this->query = '';
    }
    public function render()
    {
        return view('livewire.update');
    }
}

```

Issue:
The modal shows, the function is triggered, the model value is being updated (also in the blade component), the listbox with results is being updated, the search box is being updated, in the select the value isn't shown.

Expected behavior:
The value of the model should be selected in the options list and shown as a placeholder in the select.

Additional information:
![Image](https://github.com/user-attachments/assets/fcb096fa-0cd8-481b-8ea6-f2ea3e7487f3)

Making it a normal select (removing variant="listbox") makes it work, but you want to implement serverside search because of the many options.
The idea is to set it from outside the component but to make it easier for you to reproduce I have put it in 1 component.

**Comment by CodeAn** [comment]
Might be linked to #453  and #430 , just slightly different use case.

**Comment by calebporzio** [comment]
I pulled this down and was able to fix it by adding `wire:key` attributes to the options array.

This is essential as with any dynamic list in Livewire.

I added a new callout in the docs to hopefully emphasize this for others:

![Image](https://github.com/user-attachments/assets/0f696c92-8a95-4617-b2ba-843c3f0b6471)

**Comment by CodeAn** [comment]
Can confirm it is fixed by 1.0.15.

The wire:key is indeed necessary and was missing from my snippet because of testing.
